boardStreamer/__main__.py

```diff
@@ -5,43 +5,37 @@
 
 from functools import cached_property
 
-class CommandArgs():
-	def __init__(self):
-
-		self.parser = argparse.ArgumentParser(
-			prog='Board Streamer',
-			description='Connect to a real EEG board or simulates an EEG board to send data over a socker',
-			# epilog='Text at the bottom of help'
-		)
-
-		self.parser.add_argument(
-			'-p', '--port',
-			type=int,
-			default=800,
-			help='Port to send data to over sockets'
-		)
-
-		self.parser.add_argument(
-			'-m', '--mode',
-			choices=['sim8', 'sim16', 'com'],
-			default='com',
-			help='sets server mode'
-		)
-
-		self.parser.add_argument(
-			'-w', '--window',
-			type=int,
-			default=10,
-			help='determines the number of samples to aquire per packet'
-		)
-
-	def run(self):
-		'''
-			Returns list for keyword arguments
-		'''
-		args = self.parser.parse_args()
-
-		return vars(args)
```

Comment:
@jonathanm-12 try/catch would not solve this problem. This is because you omitted the step that transforms the Namespace to a dictionary. Please see this line you removed.

Reply:
fixed